test/OpenTelemetry.Sampler.AWS.Tests/TestMatcher.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="TestMatcher.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace OpenTelemetry.Sampler.AWS.Tests;
+
+public class TestMatcher
+{
+    [Fact]
+    public void TestWildcardMatching()
+    {
+        Assert.True(Matcher.WildcardMatch(null, "*"));
+        Assert.True(Matcher.WildcardMatch(string.Empty, "*"));
+        Assert.True(Matcher.WildcardMatch("HelloWorld", "*"));
+
+        Assert.False(Matcher.WildcardMatch(null, "Hello*"));
+        Assert.False(Matcher.WildcardMatch("HelloWorld", null));
+
+        Assert.True(Matcher.WildcardMatch("HelloWorld", "HelloWorld"));
+        Assert.True(Matcher.WildcardMatch("HelloWorld", "Hello*"));
+        Assert.True(Matcher.WildcardMatch("HelloWorld", "*World"));
+        Assert.True(Matcher.WildcardMatch("HelloWorld", "?ello*"));
```

Comment:
We don't need really complex testing here since the sampling rule options only allow for `*` and `?` for matching.
https://docs.aws.amazon.com/xray/latest/devguide/xray-console-sampling.html#xray-console-sampling-options